src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,39 @@
+import { useTodos } from '../../utils/TodoContext';
+import { TodoFilter } from '../TodoFilter';
+
+export const TodoFooter: React.FC = () => {
+  const { todos, removeTodo } = useTodos();
+  const isClearButtonActive = todos.some(todo => todo.completed);
+  const countActiveTodos = todos.filter(todo => !todo.completed).length;
+
+  const clearAllCompleted = () => {
+    todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id)
+      .forEach(id => {
+        if (typeof id === 'number') {
+          removeTodo(id);
+        }
+      });
```

Comment:
```suggestion
      .forEach(todo => {
        removeTodo(todo.id);
      });
```
Why do you need to check the id type?